### PR TITLE
optimize: Create the default values of cluster and grouplist for the custom txServiceGroup

### PR DESCRIPTION
**Pull request by wangliang181230**
### Ⅰ. Describe what this PR did
optimize: create the default values of cluster and grouplist for the custom txServiceGroup.
优化：针对用户自定义配置的`txServiceGroup`，生成`cluster`和`grouplist`配置的默认值。而不是随意添加一组默认值。

### CODE REVIEW

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java [hunk 1]
```diff
@@ -85,12 +91,37 @@ public ServiceProperties setDisableGlobalTransaction(boolean disableGlobalTransa
...
+        this.seataProperties = seataProperties;
+    }
+
     @Override
     public void afterPropertiesSet() throws Exception {
-        if (0 == vgroupMapping.size()) {
+        //Create the default cluster and grouplist for the txServiceGroup.
+        if (seataProperties != null) {
+            //The transaction service group
+            String txServiceGroup = seataProperties.getTxServiceGroup();
```
Comment (slievrly) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java:104]: if txServiceGroup is null?

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java [hunk 2]
```diff
@@ -85,12 +91,37 @@ public ServiceProperties setDisableGlobalTransaction(boolean disableGlobalTransa
         return this;
     }
 
+    @Autowired(required = false)
```
Comment (slievrly) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java:94]: Why not annotate it directly to the field？

Comment (wangliang181230) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java:94]: for the unit test

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java (hunk 1 above)
Comment (wangliang181230) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java:104]: done

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java (hunk 2 above)
Comment (wangliang181230) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java:94]: The set method is added to test cases. `@Autowired(required = false)` can also annotate it to the field.

Comment (wangliang181230) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/client/ServiceProperties.java]: I annotated it to the properties.

**Review by slievrly — APPROVED** [review]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

**Comment by wangliang181230** [comment]
@slievrly @a364176773 
#3117 这个PR中，优化的相关的异常信息打印，此PR是否不再需要了？

**Pull request closed without merging**